lib/spack/docs/packaging_guide.rst

```diff
@@ -3134,12 +3140,10 @@ File functions
 
   .. code-block:: python
 
-     install('my-header.h', join_path(prefix.include))
+     install('my-header.h', prefix.include)
 
-:py:func:`join_path(prefix, *args) <spack.join_path>`
-  Like ``os.path.join``, this joins paths using the OS path separator.
-  However, this version allows an arbitrary number of arguments, so
-  you can string together many path components.
```

Comment:
There seems to be some confusion here. Whoever wrote `join_path` was under the assumption that `os.path.join` only accepted 2 arguments, when in reality it accepts as many as you want.

I tried replacing `join_path` with `os.path.join` and discovered that the only difference between the two is that `join_path` converts every arg to a string first. At some point, we should start phasing out `join_path`, especially in Spack's core libraries. I'm fine with it in `package.py` files, but we shouldn't rely on it elsewhere.